Lib/pdb.py

```diff
@@ -1246,13 +1246,12 @@ def _getval(self, arg):
     def _getval_except(self, arg, frame=None):
         try:
             if frame is None:
-                return eval(arg, self.curframe.f_globals, self.curframe_locals)
+                return eval(arg, self.curframe.f_globals, self.curframe_locals), None
             else:
-                return eval(arg, frame.f_globals, frame.f_locals)
-        except:
-            exc_info = sys.exc_info()[:2]
-            err = traceback.format_exception_only(*exc_info)[-1].strip()
-            return _rstr('** raised %s **' % err)
+                return eval(arg, frame.f_globals, frame.f_locals), None
+        except Exception as e:
```

Comment:
Why should we only except `Exception`? The code to eval may raise anything, but it should not interrupt the main file running.

(You can use `sys.exc_info()[1]` to retrieve `e`)